src/VisualStudio/Next/FindReferences/StreamingFindReferencesPresenter.TableDataSourceFindReferencesContext.cs

```diff
@@ -358,22 +358,132 @@ private TextSpan GetRegionSpanForReference(SourceText sourceText, TextSpan refer
                     length: referenceSpan.Length);
 
                 return new TaggedTextAndHighlightSpan(taggedText, highlightSpan);
             }
 
+            private List<SymbolDisplayPart> MergeClassifiedSpans(
+                List<ClassifiedSpan> syntaxSpans, List<ClassifiedSpan> semanticSpans, 
+                TextSpan widenedSpan, SourceText sourceText)
+            {
+                // The spans produced by the language services may not be ordered
+                // (indeed, this happens with semantic classification as different
+                // providers produce different results in an arbitrary order).  Order
+                // them first before proceeding.
+                syntaxSpans = Order(syntaxSpans);
+                semanticSpans = Order(semanticSpans);
+
+                // Produce SymbolDisplayParts for both sets of ClassifiedSpans.  This will
+                // also produce parts for the regions between the sections that the classifiers
+                // returned results for (i.e. for things like spaces and plain text).
+                var syntaxParts = Classifier.ConvertClassifications(
+                    sourceText, widenedSpan.Start, syntaxSpans, insertSourceTextInGaps: true);
+                var semanticParts = Classifier.ConvertClassifications(
+                    sourceText, widenedSpan.Start, semanticSpans, insertSourceTextInGaps: true);
+
+                // Now merge the lists together, taking all the results from syntaxParts
+                // unless they were overridden by results in semanticParts.
+                return CreateAllParts(syntaxParts, semanticParts);
+            }
+
+            private List<ClassifiedSpan> Order(List<ClassifiedSpan> syntaxSpans)
+            {
+                return syntaxSpans.OrderBy(s => s.TextSpan.Start).ToList();
+            }
+
+            private List<SymbolDisplayPart> CreateAllParts(
+                List<SymbolDisplayPart> syntaxParts,
+                List<SymbolDisplayPart> semanticParts)
+            {
+                // Take all the syntax parts.  However, if any have been overridden by a 
+                // semantic part, then choose that one.
+
+                // To make life easier, determine the spans for all the parts in the lists.
+                var syntaxPartsAndSpans = AddSpans(syntaxParts);
+                var semanticPartsAndSpans = AddSpans(semanticParts);
+
+                var finalParts = new List<SymbolDisplayPart>();
+                var lastReplacementIndex = 0;
+                for (int i = 0, n = syntaxPartsAndSpans.Count; i < n; i++)
+                {
+                    var syntaxPartAndSpan = syntaxPartsAndSpans[i];
+
+                    // See if we can find a semantic part to replace this syntax part.
+                    var replacementIndex = semanticPartsAndSpans.FindIndex(
+                        lastReplacementIndex, t => t.Item2 == syntaxPartAndSpan.Item2);
```

Comment:
I audited our semantic classifiers, and all of them do nothing but semantically classify the span of an identifier.  The same span that our syntactic classifier will do.